src/libraries/System.Private.CoreLib/generators/EventSourceGenerator.Emitter.cs

```diff
@@ -12,6 +12,15 @@ namespace Generators
 {
     public partial class EventSourceGenerator
     {
+        /// <summary>Code for a [GeneratedCode] attribute to put on the top-level generated members.</summary>
+        private static readonly string s_generatedCodeAttribute = $"[global::System.CodeDom.Compiler.GeneratedCodeAttribute(\"{typeof(EventSourceGenerator).Assembly.GetName().Name}\", \"{typeof(EventSourceGenerator).Assembly.GetName().Version}\")]";
+        /// <summary>Header comments and usings to include at the top of every generated file.</summary>
+        private static readonly string[] s_headers = new string[]
```

Comment:
maybe not worth having an array you loop over unless we have more than one thing? perhaps just append the string on line 41.

Reply:
I copied it from the Regex source generator. It is more extensible in case we want to suppress warnings in the future, and isn't particularly heavy performance-wise (a `foreach` loop over an array will be translated to a simple `for` loop).

Reply:
We can still have it extensible and still using one string.

```C#

"// <auto-generated/>" + Environment.NewLine +
"any future string"           + Environment.NewLine;
```

I agree it is not perf critical though :-)

Reply:
Extensible is good when it means we can make future changes less disruptively. In this case, it's private code; we can do whatever we want. 😸 Similarly we don't have a string[] for assembly attributes here, as we just have one.

I would just append the line as you do the attribute.